packages/fuchsia_ctl/lib/src/emulator.dart

```diff
@@ -0,0 +1,232 @@
+      zbiExecutable,
+      '--compressed=zstd',
+      '-o',
+      signedZbiPath,
+      zbiPath,
+      '-e',
+      'data/ssh/authorized_keys=${authorizedKeysAbsolute.path}'
+    ];
+    await _run(zbiCommand);
+  }
+
+  /// Launch AEMU with [fvmImagePath], [signedZbiPath], and [qemuKernelPath].
+  ///
+  /// [prepareEnvironment] must have been called before starting the emulator.
+  ///
+  /// If [headless] is true, AEMU will run without a graphical window. Infra
+  /// will run AEMU in headless mode.
+  ///
+  /// [windowSize] is what AEMU will set its window size to. Defaults to
+  /// [defaultWindowSize]. Expected to be in the format of "WIDTHxHEIGHT".
+  Future<OperationResult> start(
+      {bool headless = false, String windowSize}) async {
+    assert(fvmImagePath != null && fs.isFileSync(fvmImagePath));
+    assert(signedZbiPath != null && fs.isFileSync(signedZbiPath));
+
+    final List<String> aemuCommand = <String>[
+      aemuPath,
+      '-feature',
+      'VirtioInput,RefCountPipe,KVM,GLDirectMem,Vulkan',
+      '-window-size',
+      windowSize ?? defaultWindowSize,
+      '-gpu',
+      'swiftshader_indirect',
+      if (headless) aemuHeadlessFlag,
+    ];
+
+    /// Anything after -fuchsia flag will be passed to QEMU
+    aemuCommand.addAll(<String>[
+      '-fuchsia',
+      '-kernel', qemuKernelPath,
+      '-initrd', signedZbiPath,
+      '-m', '2048',
+      '-serial', 'stdio',
+      '-vga', 'none',
+      '-device', 'virtio-keyboard-pci',
+      '-device', 'virtio_input_multi_touch_pci_1',
+      '-smp', '4,threads=2',
+      '-machine', 'q35',
+      '-device', 'isa-debug-exit,iobase=0xf4,iosize=0x04',
+      // TODO(chillers): Add hardware acceleration option to configure this.
+      '-enable-kvm',
+      '-cpu', 'host,migratable=no,+invtsc',
+      '-netdev', 'type=tap,ifname=qemu,script=no,downscript=no,id=net0',
+      '-device', 'e1000,netdev=net0,mac=52:54:00:63:5e:7a',
+      '-drive', 'file=$fvmImagePath,format=raw,if=none,id=vdisk',
+      '-device', 'virtio-blk-pci,drive=vdisk',
+      '-append',
+      // TODO(chillers): Generate entropy mixin.
+      '\'TERM=xterm-256color kernel.serial=legacy kernel.entropy-mixin=660486b6b20b4ace3fb5c81b0002abf5271289185c6a5620707606c55b377562 kernel.halt-on-panic=true\'',
+    ]);
+
+    await _start(aemuCommand);
+
+    return OperationResult.success();
+  }
+
+  /// Helper function for running [command] and manging its stdio and errors.
+  Future<void> _run(List<String> command) async {
+    stdout.writeln(command.join(' '));
+    final ProcessResult process = await processManager.run(
+      command,
+    );
+    stdout.writeln(process.stdout);
+    stderr.writeln(process.stderr);
+
+    if (process.exitCode != 0) {
+      throw EmulatorException('${command.first} did not return exit code 0');
+    }
+  }
+
+  /// Helper function for starting [command] and piping its stdio and errors.
+  Future<Process> _start(List<String> command) async {
```

Comment:
Same here, looks like a generic function that can be reused in some other places.

Reply:
Moved